state.go

```diff
@@ -643,22 +666,24 @@ func (usr *UpdateStatusReportState) Handle(ctx *StateContext, c Controller) (Sta
 		} else {
 			if usr.status == statusFailure {
 				log.Debugf("update failed, attempt log upload")
-				// TODO upload logs from the failed update, see
-				// https://tracker.mender.io/browse/MEN-437 for details
-				c.UploadLog(usr.update, []LogEntry{
-					LogEntry{
-						Timestamp: time.Now().Format(time.RFC3339),
-						Level:     "error",
-						Message:   "update failed",
-					},
-				})
+
+				logs, err := DeploymentLogger.GetLogs(usr.update.ID)
+				if err != nil {
+					log.Errorf("Failed to get deployment logs for deployment [%v]: %v",
+						usr.update.ID, err)
+				} else {
+					c.UploadLog(usr.update, logs)
```

Comment:
Might need to retry the upload a couple of times if it fails. I'm not sure how important it is to have these logs.

Actually, another thought, what if succeed in reporting the status but fail to upload the log, or why is this process split in two separate messages? Maybe it'd be better to report status _and_ upload logs at the same time, in a single message.